src/test/java/org/openrewrite/java/migrate/lang/IfElseIfConstructToSwitchTest.java

```diff
@@ -0,0 +1,496 @@
+ * You may obtain a copy of the License at
+ * <p>
+ * https://docs.moderne.io/licensing/moderne-source-available-license
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.migrate.lang;
+
+import org.junit.jupiter.api.Test;
+import org.openrewrite.DocumentExample;
+import org.openrewrite.test.RecipeSpec;
+import org.openrewrite.test.RewriteTest;
+
+import static org.openrewrite.java.Assertions.java;
+
+class IfElseIfConstructToSwitchTest implements RewriteTest {
+
+    @Override
+    public void defaults(RecipeSpec spec) {
+        spec.recipe(new IfElseIfConstructToSwitch());
+    }
+
+    @Test
+    @DocumentExample
+    void defaultSwitchBlockWithNullCheckAndFinalElseStatement() {
+        rewriteRun(
+          //language=java
+          java(
+            """
+              class Test {
+                  static String formatter(Object obj) {
+                      String formatted = "initialValue";
+                      if (obj == null) {
+                          formatted = "null";
+                      } else if (obj instanceof Integer i)
+                          formatted = String.format("int %d", i);
+                      else if (obj instanceof Long l) {
+                          formatted = String.format("long %d", l);
+                      } else if (obj instanceof Double d) {
+                          formatted = String.format("double %f", d);
+                      } else if (obj instanceof String s) {
+                          String str = "String";
+                          formatted = String.format("%s %s", str, s);
+                      } else {
+                          formatted = "unknown";
+                      }
+                      return formatted;
+                  }
+              }
+              """,
+            """
+              class Test {
+                  static String formatter(Object obj) {
+                      String formatted = "initialValue";
+                      switch (obj) {
+                          case null -> formatted = "null";
+                          case Integer i -> formatted = String.format("int %d", i);
+                          case Long l -> formatted = String.format("long %d", l);
+                          case Double d -> formatted = String.format("double %f", d);
+                          case String s -> {
+                              String str = "String";
+                              formatted = String.format("%s %s", str, s);
+                          }
+                          default -> formatted = "unknown";
+                      }
+                      return formatted;
```

Comment:
O boy, I should have seen this earlier. The outcome is still not really what I would have expected as a developer. It would be:

```suggestion
                      String formatted = switch (obj) {
                          case null -> "null";
                          case Integer i -> String.format("int %d", i);
                          case Long l -> String.format("long %d", l);
                          case Double d -> String.format("double %f", d);
                          case String s -> {
                              String str = "String";
                              return String.format("%s %s", str, s);
                          }
                          default -> "unknown";
                      }
                      return formatted;
```

(or even using the `return` statement directly) This changes the recipe quite much 😢.

Reply:
We have another ticket for this as a separate recipe. If you look at the Jep that's another topic/purpose. 
Not all of the if-else-if instance of's are assigning formatted. This is just for this test's sake. 
So in cases where the if else if construct just does stuff and not assigns the same variable, this switch statement would not have to converted to a switch expression. 
But indeed, the tests are working towards this goal as I can then reuse them all the time building on top of the outcome of a previous recipe. 
The goal would be that if you start with java 8 if-else-if construct you in the end can end up the switch expressions indeed, but by doing it over multiple recipes we guarantee that we also clean up other cases as closely as possible to this state. 
- https://github.com/openrewrite/rewrite-migrate-java/issues/718